### DISCUSSION TITLE
Will this ever support ESM?

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [nodejs/single-executable#84] [General]
Per documentation, as of now:

> The single executable application feature currently only supports running a single embedded script using the [CommonJS](https://nodejs.org/api/modules.html#modules-commonjs-modules) module system.

Is supporting ESM being considered?

**Comment by Bennch** [comment]
In combination with esbuild you can do the trick :)

**Reply by GabenGar** [reply]
That's not ESM, just a ESM input transpiled into CJS output (with all problems this entails).

**Comment by un0tec** [comment]
any updates?

**Comment by MertcanYuzer** [comment]
Thanks to v23.6.0, we have a new question: Will this ever support TypeScript?

**Reply by mkvlrn** [reply]
Yeah, been using bun [to do this](https://bun.sh/docs/bundler/executables) with proper ESM and Typescript - it is 100% painless. Any node project, just need to do a minimal bun "setup" and spit out executables.

Funny thing about this project right here is that the docs list it as [active development](https://nodejs.org/api/single-executable-applications.html#single-executable-applications), but I doubt that very very much.

**Reply by MertcanYuzer** [reply]
I think the single executable was an attempt to say, 'We are still in the race with Bun and Deno'. 
If a comparison table for runtimes were to be made, it would be enough to put a checkmark under the single executable column.
Still, I appreciate the current development. I'm using tsup to get a single-file CJS build and then building it into an executable, better than vercel/pkg I guess.
